Added databases/postgresql.md

syndbg opened the pull request:
Just something that I missed in the `sinatra-recipes`.

This is one of the few documents I have ready. Others are `data-mapper`, `activerecord` and `mysql`. But one by one.

kgrz commented on databases/postgresql.md:
````diff
@@ -0,0 +1,259 @@
+[Install PostgreSQL](http://www.postgresql.org/download/)
+
+```
+gem install pg
````
Does this need some description? something like "After the DB in installed on the machine, install the driver" etc.?

syndbg replied:
It certainly does need a description. I threw the line there without giving the reader much reason for it. I will update it !

kgrz commented on databases/postgresql.md:
````diff
@@ -0,0 +1,259 @@
+gem install pg
+```
+
+**ActiveRecord or DataMapper**. You need only one, but we'll show both.
````
Is it a good idea to split this into two separate recipes?

syndbg replied:
Sounds like a good idea. The post is too long to follow and they can just choose their preferred ORM from http://recipes.sinatrarb.com/p/databases?#article . Will update it accordingly

kgrz commented on databases/postgresql.md:
````diff
@@ -0,0 +1,259 @@
+
+**CRUD operations in routes**
+
+```ruby
````
We should probably add a `require 'models/*'` (or something similar) line here. Best to show the folder structure somewhere upfront.

syndbg replied:
Agreed! Will update it

kgrz commented on databases/postgresql.md:
````diff
@@ -0,0 +1,259 @@
+
+```
+require 'sinatra/activerecord/rake'
+...
````
extraneous line?

syndbg replied:
The require line is needed to expose the `sinatra-activerecord` gem rake functionality.
```
rake db:create
rake db:drop
rake db:migrate
..... and etc
```

kgrz commented:
Thank you so much for this!

syndbg commented:
Updated! Feel free to review it again when you have time :metal:

syndbg commented:
Ping :1234:

kgrz commented on databases/postgresql-activerecord.md:
````diff
@@ -0,0 +1,139 @@
+
+```
+▶ rake db:create_migration NAME=create_articles
+db/migrate/20150516144225_create_articles.rb
````
This looks a bit ambiguous. It looks as if we're asking to type-in the `db/migrate/...` part too. Do you mind if I fix this?

kgrz commented on databases/postgresql-activerecord.md:
```diff
@@ -0,0 +1,139 @@
+
+configure :production do
+  set :database, {adapter: 'postgres',  encoding: 'unicode', database: 'your_database_name', pool: 2, username: 'your_username', password: 'your_password'}
+end
```
Are you sure this is going to work? Doesn't AR's migration `rake` helper require the `config/database.yml` file to be present? If AR does recognize Sinatra's configuration helper, then there probably should be a specific way to load it into the `Rakefile`.

kgrz commented:
@syndbg ⏰

syndbg commented:
Alright, I really took my time. 

Just tested and it works just fine with `configure` block instead of `config/database.yml`. The tricky part was that it should be `postgresql` instead of `postgres` and in the `Rakefile` you should also require `./app`. 

Everything else is A-okay. :beetle:

kgrz commented:
Hmm. I tried setting up an app with datamapper and followed the first part of setup (without the AR-style setup) and I'm getting an error specifying that the database is not setup, and I couldn't find any instructions. Checkout this repo: https://github.com/kgrz/datamapper

If I run `require_relative 'app.rb'`, I get an error saying the database is not setup. Some more notes:

```
Sinatra AR

1. Add comments about model directory path
2. in the get index route, `@articles = Articles.all` is called. But this is supposed
   to be `Article.all`
3. Requiring `sinatra/activerecord` is a must to ensure the rake tasks
   run properly


Datamapper

1. Missing dm-migrations gem install
2. Missing instruction to create the DB. `auto_migrate!` doesn’t work
   without an existing DB.
```

syndbg commented:
@kgrz 

> Sinatra AR

> 1. Add comments about model directory path

Done.

> 2. in the get index route, `@articles = Articles.all` is called. But this is supposed to be `Article.all`

Fixed in both files.

> 3. Requiring `sinatra/activerecord` is a must to ensure the rake tasks run properly

Added.

> Datamapper

> 1. Missing dm-migrations gem install

Replaced the `dm-core` and `dm-migrations` with `data_mapper` which installs everything except database adapters, needed to use `DataMapper`.

> 2. Missing instruction to create the DB. `auto_migrate!` doesn’t work without an existing DB.

Yep, they were missing. Turns out you can't create automagically a DB with `DataMapper`.
Added instructions about that too.

syndbg commented:
@kgrz No worries. 👍
